src/read/elf/file.rs

```diff
@@ -104,6 +104,32 @@ where
             })
     }
 
+    fn notes(&self) -> impl Iterator<Item = read::Result<NoteIterator<'data, Elf>>> + '_ {
+        enum Either<L, R> {
+            Left(L),
+            Right(R),
+        }
+        let Self {
+            endian,
+            data,
+            sections,
+            segments,
+            ..
+        } = self;
+        let sections = sections.iter().map(Either::Left);
+        let segments = segments.iter().map(Either::Right);
+        std::iter::empty()
+            .chain(sections)
+            .chain(segments)
+            .filter_map(move |header| {
+                match header {
+                    Either::Left(header) => header.notes(*endian, *data),
+                    Either::Right(header) => header.notes(*endian, *data),
+                }
+                .transpose()
+            })
+    }
```

Comment:
This would work too, right?

```suggestion
        let sections = sections.iter().filter_map(|header| header.notes(*endian, *data));
        let segments = segments.iter().filter_map(|header| header.notes(*endian, *data));
        sections.chain(segments)
    }
```